src/checker.py

```diff
@@ -86,6 +86,15 @@ def _read_manifest(self, manifest_path):
         self._manifest_contents[manifest_path] = contents
         return contents
 
+    def _dump_manifest(self, manifest_path, contents):
+        _, ext = os.path.splitext(manifest_path)
+        if ext in ('.yaml', '.yml'):
+            raise NotImplementedError(
+                "Updating YAML manifests is not yet supported"
+            )
+        else:
+            return json.dumps(contents, indent=4)
```

Comment:
AFAIK this may end up giving you slightly a different JSON file than the input (beyond what are the changes). e.g. comments will be stripped of, possibly also the formatting and/or sorting of elements.
These are the main reasons why I didn't implement this solution from the beginning, but even with thos eventual annoyances, it's good to see this feature in now! Kudos!

Reply:
It actually does preserve element ordering in dicts due to [this](https://github.com/endlessm/flatpak-external-data-checker/blob/master/src/checker.py#L63)¹, but otherwise, yes, it normalises the formatting (which I consider a feature :stuck_out_tongue:) and removes comments (which are useful, sure, but personally I can live without them in exchange for this).

¹ Actually, from Python 3.6, `dict` is ordered, so `json.dumps(json.loads(s))` never reorders keys even if you don't specify `object_pairs_hook`.

Reply:
Awesome! Thanks for clarifying. I didn't see an answer to my other comment, so I am keeping this unmerged and let you decide.